src/data/mv_polynomial.lean

```diff
@@ -1407,6 +1418,54 @@ lemma pderivative_zero {v : S} : pderivative v (0 : mv_polynomial S R) = 0 :=
 suffices pderivative v (C 0 : mv_polynomial S R) = 0, by simpa,
 show pderivative v (C 0 : mv_polynomial S R) = 0, from pderivative_C
 
+section
+variables (R)
+
+/-- `pderivative : S → mv_polynomial S R → mv_polynomial S R` as an `add_monoid_hom`  -/
+def pderivative.add_monoid_hom (v : S) : mv_polynomial S R →+ mv_polynomial S R :=
+{ to_fun := pderivative v,
+  map_zero' := pderivative_zero,
+  map_add' := λ x y, pderivative_add, }
+
+@[simp]
+lemma pderivative.add_monoid_hom_apply (v : S) (p : mv_polynomial S R) :
+  (pderivative.add_monoid_hom R v) p = pderivative v p :=
+rfl
+end
+
+private lemma pderivative_eq_zero_of_not_mem_vars_aux {v : S} {f : mv_polynomial S R}
+  (h : v ∉ vars f) (x : S →₀ ℕ) (H : x ∈ f.support) :
+  x v = 0 :=
```

Comment:
Let's move this lemma to the section where we define `vars` in the first place. The only trick will be to name the lemma (which shouldn't be private). How about `mem_support_not_mem_vars_zero`? Perhaps also change the order of arguments to:
```
{f : mv_polynomial S R} {x : S →₀ ℕ} (H : x ∈ f.support) {v : S} (h : v ∉ vars f) 
```